socs/agents/scpi_psu/agent.py

```diff
@@ -188,15 +351,22 @@ def main(args=None):
                                   parser=parser,
                                   args=args)
 
+    init_params = False
+    if args.mode == 'acq':
+        init_params = {'auto_acquire': True}
     agent, runner = ocs_agent.init_site_agent(args)
 
-    p = ScpiPsuAgent(agent, args.ip_address, int(args.gpib_slot))
+    p = ScpiPsuAgent(agent, args.ip_address, int(args.gpib_slot), port=int(args.port))
```

Comment:
This'll hit a `TypeError` when either `--gpib-slot` or `--port` are not passed:
```
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType
```

This is what the tests are failing on, as they don't supply `--port`, only `--gpib-port`.

You could move the cast into the agent's `__init__` after the check for `None`.

Reply:
fixed